modules/gapi/include/opencv2/gapi/stereo.hpp

```diff
@@ -0,0 +1,92 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distereoibution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifndef OPENCV_GAPI_STEREO_HPP
+#define OPENCV_GAPI_STEREO_HPP
+
+#include <opencv2/calib3d.hpp>
+
+#include <opencv2/gapi/gmat.hpp>
+#include <opencv2/gapi/gscalar.hpp>
+#include <opencv2/gapi/gkernel.hpp>
+
+
+namespace cv {
+namespace gapi {
+
+enum class StereoOutputFormat {
+    DISPARITY,
+    DEPTH
+};
+
+/** @brief Structure for the Stereo operation initialization parameters.*/
+struct GAPI_EXPORTS StereoInitParam {
+    int numDisparities;
+    int blockSize;
+    double baseline;
+    double focus;
+};
+
+namespace calib3d {
+
+/** @brief Structure for the Stereo operation setup parameters.*/
+struct GAPI_EXPORTS StereoSetup {
+    double baseline;
+    double focus;
+    cv::Ptr<cv::StereoBM> stereoBM;
+};
+
+G_TYPED_KERNEL(GStereo, <GMat(GMat, GMat, const cv::gapi::StereoOutputFormat, const int)>, "org.opencv.stereo") {
+    static GMatDesc outMeta(GMatDesc ly, GMatDesc ry,
+                            const cv::gapi::StereoOutputFormat oF,
+                            const int  dtype) {
+
+        GAPI_Assert(ly.chan == 1);
+        GAPI_Assert(ly.depth == CV_8U);
+
+        GAPI_Assert(ry.chan == 1);
+        GAPI_Assert(ry.depth == CV_8U);
+
+        GAPI_Assert(oF == cv::gapi::StereoOutputFormat::DEPTH ||
+                    oF == cv::gapi::StereoOutputFormat::DISPARITY);
+
+        GAPI_Assert(dtype == CV_16U || dtype == CV_16F);
+
+        return dtype == CV_16U ? ly.withDepth(CV_16U) : ly.withDepth(CV_16F);
+    }
+};
+
+} // namespace calib3d
+
+/** @brief Extract disparity/depth information depending on passed StereoOutputFormat argument.
+The function extracts disparity/depth information depending on passed StereoOutputFormat argument from
+given stereo-pair and convert output information to the type specified in the dtype.
+
+@param ly Y plane from left camera: 8-bit unsigned 1-channel matrix of @ref CV_8UC1 type
+@param ry Y plane from right camera: 8-bit unsigned 1-channel matrix of @ref CV_8UC1 type
+@param oF the enum class to specify to make disparity to depth conversion or not
+@param dtype is a type of output @ref cv::Mat. Can be @ref CV_16U or @ref CV_16F
+*/
+
+GAPI_EXPORTS GMat stereo(const GMat& ly,
+                         const GMat& ry,
+                         const cv::gapi::StereoOutputFormat oF = cv::gapi::StereoOutputFormat::DISPARITY,
+                         const int dtype = CV_FP16);
+
+} // namespace gapi
+
+namespace detail {
+
+template<> struct CompileArgTag<cv::gapi::StereoInitParam> {
+    static const char* tag() {
+        return "org.opencv.stereoInit";
+    }
+};
```

Comment:
Who needs that and why?

Reply:
Its needs for take `StereoInit` from `CompileArgs`